HW_8/task 8.2.js

```diff
@@ -0,0 +1,269 @@
+        id: 3,
+        name: 'Отдел маркетинга',
+        employees_count: 20,
+      },
+      {
+        id: 4,
+        name: 'Администрация',
+        employees_count: 15,
+      },
+    ],
+  },
+  {
+    id: 5,
+    name: 'Предприятие 2',
+    departments: [
+      {
+        id: 6,
+        name: 'Отдел разработки',
+        employees_count: 50,
+      },
+      {
+        id: 7,
+        name: 'Отдел маркетинга',
+        employees_count: 20,
+      },
+      {
+        id: 8,
+        name: 'Отдел охраны труда',
+        employees_count: 5,
+      },
+    ],
+  },
+  {
+    id: 9,
+    name: 'Предприятие 3',
+    departments: [
+      {
+        id: 10,
+        name: 'Отдел аналитики',
+        employees_count: 0,
+      },
+    ],
+  },
+];
+
+// Задания:
+// 1. Вывести все предприятия и их отделы. Рядом указать количество сотрудников.
+// Для предприятия посчитать сумму всех сотрудников во всех отделах.
+
+// **Пример:**
+
+// Предприятие 1 (45 сотрудников)
+// - Отдел тестирования (10 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Администрация (15 человек)
+// Предприятие 2 (75 сотрудников)
+// - Отдел разработки (50 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Отдел охраны труда (5 сотрудников)
+// Предприятие 3 (нет сотрудников)
+// - Отдел аналитики (нет сотрудников)
+
+const enterpriseStructure = (enterprises) => {
+    enterprises.map(enterprise => {
+        const totalEmployees = enterprise.departments.reduce((acc, department) => acc + department.employees_count, 0);
+        if (/.*1$/.test(totalEmployees) && !/.*11$/.test(totalEmployees)) {
+            console.log(`${enterprise.name} (${totalEmployees} сотрудник)`);
+        } else {
+        console.log(`${enterprise.name} (${totalEmployees} сотрудников)`);
+        }
+        enterprise.departments.map(department => {
+            if (/.*1$/.test(department.employees_count) && !/.*11$/.test(department.employees_count)) {
+                console.log(`- ${department.name} (${department.employees_count} сотрудник)`);
+            } else {
+            console.log(`- ${department.name} (${department.employees_count} сотрудников)`);
+            }
+        });
+    });
+}
+
+enterpriseStructure(enterprises);
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+// Функция-помощник для получения предприятия по id отдела или названию отдела:
+const getEnterpriseByDepartment = (idOrName) => enterprises.filter((enterprise) => enterprise.departments.find((department) => department.id === idOrName || department.name === idOrName));
+
+const getEnterpriseName = (idOrName) => {
+  const enterpriseFound = getEnterpriseByDepartment(idOrName);
+  if (enterpriseFound == 0) throw new Error('Нет предприятия с данным id отдела или названием отдела');
```

Comment:
According to getEnterpriseByDepartment function, it returns an array of enterprises. So enterpriseFound is an array, and array never == 0) length can be 0

Reply:
I used `==` to convert and compare operands with the different types, but it's a bad practise. I'll fix it using your suggestion